sabnzbd/utils/getperformance.py

```diff
@@ -1,14 +1,39 @@
+import platform, subprocess
+
+
 def getcpu():
-    # On Linux, let's get the CPU model name:
+    # find the CPU name (which needs a different method per OS), and return it
+    # return None if none found
+    # works on Linux, MacOS (aka OSX), Windows
+
     cputype = None
+
     try:
-        for myline in open("/proc/cpuinfo"):
-            if myline.startswith(('model name')):
-                cputype = myline[13:].rstrip()
-                break
+        if platform.system() == "Windows":
+            import _winreg as winreg	# needed on Python 2
+            key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, r"Hardware\Description\System\CentralProcessor\0")
+            cputype = winreg.QueryValueEx(key, "ProcessorNameString")[0]
+            winreg.CloseKey(key)
+
+        elif platform.system() == "Darwin":
+            cputype = subprocess.check_output(['sysctl', "-n", "machdep.cpu.brand_string"]).strip()
+
+        elif platform.system() == "Linux":
+            for myline in open("/proc/cpuinfo"):
+                if myline.startswith(('model name')):
+                    # Typical line:
+                    # model name      : Intel(R) Xeon(R) CPU           E5335  @ 2.00GHz
+                    cputype = myline.split(":", 1)[1]	# get everything after the first ":"
+                    break # we're done
+
     except:
-        # probably not on Linux
+        # An exception, maybe due to a subprocess call gone wrong
+        cputype = "Exception happened"
```

Comment:
IMHO: No. It should never happen, and so it should never appear. And if it happens I want to know it so that we know where to look for the problem. So a king of assert.
It's a different case than on 'unknown' OSes; then the output is None, and nothing appears.